SonarQube.Client/Models/SonarQubeIssue.cs

```diff
@@ -93,6 +93,12 @@ public IssueLocation(string filePath, string moduleKey, IssueTextRange textRange
         public string ModuleKey { get; }
         public IssueTextRange TextRange { get; }
         public string Message { get; }
+
+        /// <summary>
+        /// Note: currently the hash for secondary locations is calculated
+        /// post-construction, so we need to be able to update this property
+        /// </summary>
+        public string Hash { get; internal set; }
```

Comment:
As discussed previously, we could either update the location hash in-place, or build a completely new Issue-Flow-Location structure with the new data.
Updating in place is much simpler so I went with that.